tests/test_faster_rcnn.py

```diff
@@ -0,0 +1,377 @@
+import argparse
+import os
+from unittest.mock import MagicMock, patch
+
+import numpy as np
+import pytest
+import torch
+import torchvision
+from PIL import Image
+from tensorboardX import SummaryWriter
+from yacs.config import CfgNode as CN
+
+from datasetinsights.data.dummy.dummy_object_detection import DummyDetection2D
+from datasetinsights.estimators import Estimator
+from datasetinsights.estimators.faster_rcnn import (
+    collate_fn,
+    create_dataloader,
+    create_dryrun_dataset,
+    create_optimizer_lrs,
+    create_sampler,
+    download_data,
+    get_transform,
+    load_data,
+)
+from datasetinsights.storage.checkpoint import create_checkpointer
+
+TRAIN = "train"
+VAL = "val"
+TEST = "test"
+
+label_mappings = {"1": "car", "2": "bike"}
+dummy_data = DummyDetection2D(
+    transform=get_transform(), label_mappings=label_mappings
+)
+
+
+@pytest.fixture
+def config():
+    parser = argparse.ArgumentParser(
+        description="Datasetinsights Modeling Interface"
+    )
+    parser.add_argument(
+        "--config", type=str, required=True, help="config file for this model",
+    )
+    cmd_args = parser.parse_args(
+        ["--config=tests/configs/faster_rcnn_groceries_real_test.yaml"]
+    )
+    cfg = CN.load_cfg(open(cmd_args.config, "r"))
+
+    return cfg
+
+
+@patch("datasetinsights.estimators.faster_rcnn.Dataset.create")
+def test_faster_rcnn_train_one_epoch(mock_create, config):
+    mock_create.return_value = dummy_data
+    config.system.logdir = "/tmp/train_one/"
```

Comment:
This path is os dependent. I will suggest using `tempdir.TemporartDirectory()`. This method will give you different temporary directory everytime you run a single unit of unittest.

Reply:
Fixed